### PR TITLE
move logic of prepareBuild to setBuildName

### PR DESCRIPTION
**Pull request by ilya-varchenya**

### CODE REVIEW

File: src/com/qaprosoft/jenkins/pipeline/runner/AbstractRunner.groovy [hunk 1]
```diff
@@ -10,10 +10,86 @@ import static com.qaprosoft.jenkins.pipeline.Executor.*
...
+        Configuration.remove("device")
+        //TODO: move it to the global jenkins variable
+        Configuration.set("capabilities.newCommandTimeout", "180")
+        Configuration.set("java.awt.headless", "true")
+    }
 
     public AbstractRunner(context) {
         super(context)
         initOrganization()
+        setBuildName()
```
Comment (ilya-varchenya) [inline src/com/qaprosoft/jenkins/pipeline/runner/AbstractRunner.groovy:92]: do we need to call set-method here or in definite classes?

File: src/com/qaprosoft/jenkins/pipeline/runner/AbstractRunner.groovy [hunk 2]
```diff
@@ -10,10 +10,86 @@ import static com.qaprosoft.jenkins.pipeline.Executor.*
 public abstract class AbstractRunner extends BaseObject {
     // organization folder name of the current job/runner
     protected String organization = ""
+    protected String buildNameTemplate = "${BUILD_NUMBER}"
+    protected final String BUILD_NAME_SEPARATOR = "|"
+
+    public String getBuildName() {
+        return this.buildNameTemplate
```
Comment (vdelendik) [inline src/com/qaprosoft/jenkins/pipeline/runner/AbstractRunner.groovy:17]: getBuildName != getting buildNameTemplate;)

File: src/com/qaprosoft/jenkins/pipeline/runner/maven/TestNG.groovy [hunk 3]
```diff
@@ -79,6 +79,7 @@ public class TestNG extends Runner {
     public TestNG(context, jobType) {
         this (context)
         this.jobType = jobType
+        setBuildName("${BUILD_NUMBER}${suite}${branch}${env}${browser}${browserVersion}${locale}${language}${node}")
```
Comment (vdelendik) [inline src/com/qaprosoft/jenkins/pipeline/runner/maven/TestNG.groovy:82]: that's a call to update buildTemplateName so setter/getter should be named accordingly
setBuildNameTemplate(...)

File: src/com/qaprosoft/jenkins/pipeline/runner/AbstractRunner.groovy [hunk 4]
```diff
@@ -10,10 +10,52 @@ import static com.qaprosoft.jenkins.pipeline.Executor.*
...
+        String branch = Configuration.get("branch")
+        String env = Configuration.get("env")
+        String browser = getBrowser()
+        String browserVersion = getBrowserVersion()
+        String locale = Configuration.get("locale")
+        String language = Configuration.get("language")
+        String node = Configuration.get("node")
+
+        context.stage('Preparation') {
+            buildNameTemplate = "#${buildNumber}|${suite}|${branch}"
```
Comment (vdelendik) [inline src/com/qaprosoft/jenkins/pipeline/runner/AbstractRunner.groovy:33]: goal of the ticket was to replaced items from template if any...
also think about using resolveVars approach as we did in configuration.groovy. The only problem howto replaces multiply "|" symbols after resolving...

File: src/com/qaprosoft/jenkins/pipeline/runner/maven/TestNG.groovy [hunk 5]
```diff
@@ -689,85 +725,6 @@ public class TestNG extends Runner {
         return Configuration.get("node")
     }
 
-    //TODO: moved almost everything into argument to be able to move this methoud outside of the current class later if necessary
```
Comment (vdelendik) [inline src/com/qaprosoft/jenkins/pipeline/runner/maven/TestNG.groovy:692]: that's important TODO message! Almost everything means something to keep!
Without this method you lost prepareForMobile at all!

File: src/com/qaprosoft/jenkins/pipeline/runner/maven/TestNG.groovy [hunk 6]
```diff
@@ -689,85 +725,6 @@ public class TestNG extends Runner {
...
-        // ATTENTION! Obligatory remove device from the params otherwise
-        // hudson.remoting.Channel$CallSiteStackTrace: Remote call to JNLP4-connect connection from qpsinfra_jenkins-slave_1.qpsinfra_default/172.19.0.9:39487
-        // Caused: java.io.IOException: remote file operation failed: /opt/jenkins/workspace/Automation/<JOB_NAME> at hudson.remoting.Channel@2834589:JNLP4-connect connection from
-        Configuration.remove("device")
-        //TODO: move it to the global jenkins variable
-        Configuration.set("capabilities.newCommandTimeout", "180")
-        Configuration.set("java.awt.headless", "true")
-    }
-
-    protected void prepareForAndroid() {
```
Comment (vdelendik) [inline src/com/qaprosoft/jenkins/pipeline/runner/maven/TestNG.groovy:757]: where is prepareForAndroid and prepareForiOS now?

File: src/com/qaprosoft/jenkins/pipeline/runner/maven/TestNG.groovy [hunk 7]
```diff
@@ -730,7 +700,7 @@ public class TestNG extends Runner {
 
     protected void prepareForMobile() {
         logger.info("Runner->prepareForMobile")
-        def devicePool = Configuration.get("devicePool")
+        def devicePool = Configuration.get("devicePool") //this parameter is not using
```
Comment (ilya-varchenya) [inline src/com/qaprosoft/jenkins/pipeline/runner/maven/TestNG.groovy:703]: Do we need this parameter or I can delete it?

File: src/com/qaprosoft/jenkins/pipeline/runner/maven/TestNG.groovy [hunk 8]
```diff
@@ -112,7 +111,7 @@ public class TestNG extends Runner {
 
 			try {
 				getScm().clone(true)
-
+                setDisplayNameTemplate('#${BUILD_NUMBER}|${suite}|${branch}|${env}|${browser}|${browserVersion}|${locale}|${language}')
```
Comment (vdelendik) [inline src/com/qaprosoft/jenkins/pipeline/runner/maven/TestNG.groovy:114]: Please move it by default to constructor if possible

File: src/com/qaprosoft/jenkins/pipeline/runner/maven/TestNG.groovy (hunk 7 above)
Comment (ilya-varchenya) [reply src/com/qaprosoft/jenkins/pipeline/runner/maven/TestNG.groovy:707]: Created ticket https://github.com/qaprosoft/qps-pipeline/issues/817

**Pull request merged**